test/e2e/parallel/nstemplatetier_test.go

```diff
@@ -231,7 +231,7 @@ func verifyResourceUpdatesForUserSignups(t *testing.T, hostAwait *wait.HostAwait
 		require.NoError(t, err)
 		require.NotNil(t, userAccount)
 
-		nsTemplateSet, err := memberAwaitility.WaitForNSTmplSet(t, usersignup.Status.CompliantUsername, wait.UntilNSTemplateSetHasTier(tier.Name))
+		nsTemplateSet, err := memberAwaitility.WaitForNSTmplSet(t, usersignup.Status.CompliantUsername, wait.UntilNSTemplateSetHasTier(tier.Name), tiers.UntilNSTemplateSetHasStatusTemplateRefs(tiers.GetTemplateRefs(t, hostAwait, tier.Name)))
```

Comment:
correct me if I'm wrong, this criteria could be applied to all NSTemplateSet CRs, is that correct?
If yes, then why not moving it either here:
https://github.com/codeready-toolchain/toolchain-e2e/blob/63ca31925d122fddf659817e147bbc74baaaa370/testsupport/user_assertions.go#L263-L266
or here:
https://github.com/codeready-toolchain/toolchain-e2e/blob/63ca31925d122fddf659817e147bbc74baaaa370/testsupport/tiers/nstemplateset.go#L22-L24

so it can be verified for all provisioned NSTemplateSets

Reply:
I did that initially see: https://github.com/codeready-toolchain/toolchain-e2e/pull/1137/commits/9f4e208cfd5d72a15b1076523ed4e853a9fe42e7

but it will fail on the migration tests, since currently the field is not there. 

We can move it there once these PRs are merged maybe ?

Reply:
probably something to fix in the migration tests. Do you remember what was the failure?

Reply:
the call stack is: 
`verifyAppStudioProvisionedSignup` in `verify_migration_test.go` -> `VerifyResourcesProvisionedForSignupWithTiers` -> `VerifySpaceRelatedResources` in `user_assertions.go` -> `VerifyNSTemplateSet`

also `VerifyResourcesProvisionedForSignup` goes over the same chain of calls and it is used in multiple parts of the `verify_migration_test.go` if I'm seeing this correctly.

Reply:
hmmm... the fact that it's being called in the `verify_migration_test.go` is correct (I thought that you were talking about the setup part), the existing NSTemplateSets should be all updated with the references in the status. If this didn't happen, then there was a bug.

Reply:
> the existing NSTemplateSets should be all updated with the references in the status. If this didn't happen, then there was a bug.

but the failure was when checking the NSTemplates sets deployed using the current `master` version of the code, which is expected to not have the status field.

Reply:
right, the accounts were provisioned with the master version. But the version from your PR was deployed after that (before running `verify_migration_test.go`) and the controller reconciled over all NSTemplateSet CRs present in the cluster, and it should update the status in all of them.

Reply:
ok, not sure why I was convinced that we run the `verify_migration_tests.go` also after deploying the master version. Than maybe there was a glitch or there is a problem. 

I can try and move it back. Or we can merge it as is and I can open a PR to move it back and see what happens.

Reply:
Let's move it back in this PR, to verify that the migration really works as expected and all NSTemplateSets will get updated

Reply:
ok, let's see. done in https://github.com/codeready-toolchain/toolchain-e2e/pull/1137/commits/88a82517979cda410dc308ef7adf375a1688b1d4

Reply:
it looks like that it worked - e2e tests are green :slightly_smiling_face:

Reply:
Yeah not sure, maybe it was a PEBKAC 🤦‍♂️ 

Thanks for the suggestion. I'll run again the e2e tests in host operator just to be sure, and I'll merge it.